xla/service/gpu/runtime/annotation.cc

```diff
@@ -0,0 +1,216 @@
+#include "xla/service/gpu/runtime/annotation.h"
+
+#include "xla/hlo/ir/dfs_hlo_visitor_with_default.h"
+
+namespace xla {
+namespace gpu {
+
+namespace {
+nvtxStringHandle_t registerString(const char* str) {
+#if GOOGLE_CUDA
+  auto domain = tsl::profiler::nvtx::GetNVTXDomain();
+  if (!domain) {
+    // NVTX not enabled, so don't bother registering strings with it
+    return {};
+  }
+  std::string buffer{};
+  constexpr auto max_length = 65330;
+  if (auto const length = std::strlen(str); length >= max_length) {
+    // nvbugs 4340868
+    std::string_view suffix{"\n[truncated]\n"};
+    buffer.reserve(max_length);
+    buffer.assign(str, str + length - suffix.size());
```

Comment:
I kept staring at this and I know `.assign()` is way overloaded, but did you mean
```suggestion
    buffer.assign(str, length - suffix.size());
```
instead? Corresponding to (5) here: https://en.cppreference.com/w/cpp/string/basic_string/assign.

Reply:
I believe your suggestion would also be correct, but that the existing code just uses overload (7) to get the same result.